src/components/Button/Button.tsx

```diff
@@ -1,13 +1,90 @@
-import { Text, TouchableOpacity } from 'react-native'
+import React from 'react'
+import { MaterialIcons } from '@expo/vector-icons'
+import { View } from 'react-native'
+import {
+  Button as PaperButton,
+  ButtonProps as PaperButtonProps,
+} from 'react-native-paper'
+import { match } from 'ts-pattern'
 
-import { ButtonProps } from './Button.types'
+type IconProps =
+  | {
+      materialIconName?: keyof typeof MaterialIcons.glyphMap
+      icon?: undefined
+    }
+  | {
+      materialIconName?: undefined
+      icon?: React.ReactNode
+    }
+type ButtonProps = Omit<PaperButtonProps, 'icon' | 'children'> & {
+  onPress?: () => void
+  alignIcon?: 'left' | 'right'
+  loading?: boolean
+} & IconProps
+
+export default function Button({
+  icon,
+  materialIconName,
+  alignIcon = 'left',
+  loading = false,
+  disabled,
+  onPress,
+  children,
+  mode = 'contained',
+  ...rest
+}: React.PropsWithChildren<ButtonProps>) {
+  const matchValues = {
+    icon: Boolean(icon || materialIconName),
+    children: Boolean(children),
+  }
+
+  const iconGap = match(matchValues)
+    .with({ icon: true, children: false }, () => -16)
+    .with({ icon: true, children: true }, () => 8)
+    .with({ icon: false, children: true }, () => 0)
+    .otherwise(() => 0)
 
-const Button = ({ label, onPress }: ButtonProps) => {
   return (
-    <TouchableOpacity onPress={onPress}>
-      <Text>{label}</Text>
-    </TouchableOpacity>
+    <View style={{ paddingHorizontal: 20, width: '100%' }}>
+      <PaperButton
+        {...rest}
+        disabled={loading || disabled}
+        icon={() =>
+          match({ icon, materialIconName })
+            .with({ materialIconName: undefined }, () => icon)
+            .with({ icon: undefined }, () => (
+              <MaterialIcons
+                name={materialIconName}
+                size={20}
+                color={mode == 'outlined' ? '#000' : '#fff'}
+              />
+            ))
+            .otherwise(() => undefined)
```

Comment:
Acho que aqui seria um bom lugar pra usar os [patterns](https://github.com/gvergnaud/ts-pattern#patterns) do ts-pattern. Daí podemos testar pelo que queremos mostrar ao invés da ausência do outro componente. Acho que fica mais claro o objetivo e mais fácil de alterarmos as coisas no futuro.
```suggestion
          match({ icon, materialIconName })
            .with({ icon: P.not(P.nullish(icon)) }, () => icon)
            .with({ materialIconName: P.not(P.nullish(materialIconName)) }, () => (
              <MaterialIcons
                name={materialIconName}
                size={20}
                color={mode == 'outlined' ? '#000' : '#fff'}
              />
            ))
            .otherwise(() => undefined)
```